src/streams/register.rs

```diff
@@ -108,7 +112,11 @@ impl ReceiverRegister {
         if did_wrap {
             // The indices wrapped around the end of `packet_status`, we clear any entries between
```

Comment:
now I am thinking maybe we don't need to clear the entries. Since we already clear them as we move `nack_check_from` ahead, and we don't allow updating positions before..